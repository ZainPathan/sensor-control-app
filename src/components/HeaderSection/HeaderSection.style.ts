import { makeStyles } from '@material-ui/core';


const useHeaderSectionStyles = makeStyles({
    root: {
        flexGrow: 1,
        minHeight: 50,
    },
    appBar: {
        padding: 10,
    }
});

export default useHeaderSectionStyles;