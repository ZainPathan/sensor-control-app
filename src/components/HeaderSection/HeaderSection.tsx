import React, { useState } from 'react';
import {
    AppBar,
    Grid,
    Switch,
    Typography
} from '@material-ui/core';
import useHeaderSectionStyles from './HeaderSection.style';

interface HeaderSectionProps {
    handleSensorToggle: Function
}

const HeaderSection: React.FC<HeaderSectionProps> = ({handleSensorToggle}) => {
    const classes = useHeaderSectionStyles();
    const [isShowAllSensors, setShowAllSensorsFlag] = useState(true);

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setShowAllSensorsFlag(event.target.checked);
        handleSensorToggle(event.target.checked);
    }

    return (
        <div className={classes.root}>
            <AppBar
                position="static"
                className={classes.appBar}
            >
                <Grid container
                      alignItems="center"
                      justify="space-between"
                      spacing={2}>
                    <Grid item>
                        <Typography variant="h6" color="inherit">
                            Sensor Management
                        </Typography>
                    </Grid>
                    <Grid item>
                        <Grid component="label"
                              container
                              alignItems="center"
                              spacing={1}
                        >
                            <Grid item>Show Connected</Grid>
                            <Grid item>
                                <Switch
                                    checked={isShowAllSensors}
                                    onChange={handleChange}
                                    value="checked"
                                />
                            </Grid>
                            <Grid item>Show All</Grid>
                        </Grid>
                    </Grid>
                </Grid>
            </AppBar>
        </div>
    );
}

export default HeaderSection;