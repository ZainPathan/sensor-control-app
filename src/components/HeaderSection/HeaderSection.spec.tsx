import React from 'react';
import renderer from 'react-test-renderer';
import HeaderSection from './HeaderSection';
import {fireEvent, render} from "@testing-library/react";

describe('HeaderSection tests', () => {
    const mockSensorToggleFn = jest.fn();

    it('snapshot test', () => {
        const tree = renderer
            .create(
                <HeaderSection
                    handleSensorToggle={mockSensorToggleFn}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with App title', () => {
        const appTitleText = 'Sensor Management';
        const { getByText } = render(
            <HeaderSection
                handleSensorToggle={mockSensorToggleFn}
            />
        );
        expect(getByText(appTitleText)).toBeTruthy();
    });

    it('toggle handler function is called', () => {
        const { getByRole } = render(
            <HeaderSection
                handleSensorToggle={mockSensorToggleFn}
            />
        );
        fireEvent.click(getByRole('checkbox'));
        expect(mockSensorToggleFn).toHaveBeenCalled();
    });
})