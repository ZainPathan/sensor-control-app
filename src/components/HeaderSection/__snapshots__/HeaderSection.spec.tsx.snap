// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection tests snapshot test 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-2 MuiPaper-elevation4"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
        >
          Sensor Management
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <label
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            Show Connected
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <span
              className="MuiSwitch-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-4 Mui-checked"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={true}
                    className="PrivateSwitchBase-input-6 MuiSwitch-input"
                    onChange={[Function]}
                    type="checkbox"
                    value="checked"
                  />
                  <span
                    className="MuiSwitch-thumb"
                  />
                </span>
              </span>
              <span
                className="MuiSwitch-track"
              />
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            Show All
          </div>
        </label>
      </div>
    </div>
  </header>
</div>
`;
